core/vcpu.c

```diff
@@ -4311,6 +4316,66 @@ static void vcpu_state_dump(struct vcpu_t *vcpu)
             vcpu->state->_idt.base,
             vcpu->state->_idt.limit,
             (uint32_t)vcpu->state->_efer);
+
+    // Dump FPU state
+    hax_log(HAX_LOGW,
+            "FX layout:\n"
+            "fcw:    %08x  fsw:    %08x  ftw: %08x  res1: %08x  fop: %08x\n"
```

Comment:
1. Uppercase each state as vcpu state format.
2. Pay attention to the space usage. Use one space after the colon and use 2 spaces between states.

Reply:
1. OK
2. I use 2 spaces between states. Field names are of different length, so I use different number  of spaces after the colon, so the output of different lines will be aligned by field names start and values, and that makes the output easier to understand. Are you sure to use one space? See how the output looks like, I think it is better.

Reply:
Great, using column alignment is better. I agree with you.